drivers/serial/uart_ns16550.c

```diff
@@ -367,7 +371,24 @@ static int uart_ns16550_configure(const struct device *dev,
 	}
 #endif
 
-	set_baud_rate(dev, cfg->baudrate);
+	/*
+	 * set clock frequency from clock_frequency property if valid,
+	 * otherwise, get clock frequency from clock manager
+	 */
+	if (dev_cfg->sys_clk_freq != 0U) {
+		pclk = dev_cfg->sys_clk_freq;
+	} else {
+		clk_dev = device_get_binding(dev_cfg->clock_drv);
```

Comment:
We are moving away from `device_get_binding()` to `DEVICE_DT_GET()`. You can refer to commit 93f2fedc12c0ad4e2e666415a498b407fcbaa3c6 on similar changes related to clock control.

Reply:
ok. I will make the change accordingly